stdlib/public/core/Codable.swift

```diff
@@ -89,6 +89,17 @@ extension CodingKey {
   public var debugDescription: String {
     return description
   }
+
+  /// A simplified description: the int value, if present, in square brackets.
+  /// Otherwise, the string value by itself. Used when concatenating coding keys
+  /// to form a path when printing debug information.
+  var errorPresentationDescription: String {
+    if let intValue {
+      "[\(intValue)]"
```

Comment:
I'm starting to wonder whether we actually need the brackets. It's not like you'll ever be in a situation where you need help differentiating between keys and indices. This isn't PHP, for cryin' out loud. Thoughts on turning `[0]` into `0`?